### ISSUE TITLE
Add validation-side negative-frame metrics (val/loss_negative, val/n_negative)

### ISSUE THREAD

**Issue by talmo** [talmolab/sleap-nn#577]
## Summary

Negative-frame training support (PR #484, shipped in v0.1.3) logs **train-side** split
metrics but no **validation-side** ones. There is no `val/loss_negative` /
`val/n_negative`, so users cannot see how the model performs on background/negative
frames in the validation set — which is exactly the false-positive signal the feature
exists to improve.

## Background

`data_config.use_negative_frames` (added in PR #484) includes user-marked negative
frames as background training examples. Negative frames participate in the train/val
split like positive frames.

The training path logs, via `_compute_negative_weighted_loss()` in
`sleap_nn/training/lightning_modules.py`:

- `train/loss_positive`
- `train/loss_negative`
- `train/n_positive`
- `train/n_negative`

But every `validation_step` (`SingleInstanceModel`, `CentroidModel`, `BottomUpModel`,
`BottomUpMultiClassModel`) uses a plain `nn.MSELoss()` with no positive/negative split
— so negatives are only folded into an undifferentiated `val/loss`. There is no
`val/loss_negative`, `val/loss_positive`, `val/n_negative`, or `val/n_positive`.

## Why it matters

- The point of negative frames is reducing false positives on empty frames. The
  **validation** loss on negatives is the metric that tells a user whether that is
  working — and it is not exposed.
- The train/val asymmetry is confusing: users see `train/loss_negative` trend but have
  no val counterpart to check for overfitting on negatives.
- Downstream: the SLEAP GUI negative-frames work (talmolab/sleap#640) wants to surface
  these in the training monitor; with no val metric there is nothing to show for the
  validation split.

## Related secondary gap — CSV logger

The train-side negative metrics that *do* exist are **not** written to
`training_log.csv`. `csv_log_keys` (built in `sleap_nn/training/model_trainer.py`,
~lines 895-942) lists only `epoch, train/loss, val/loss, learning_rate, train/time,
val/time` plus per-head keys. The `*_negative` / `*_positive` / `n_*` keys are omitted.
They reach ZMQ and WandB/TensorBoard but not the CSV log. Worth fixing in the same
change.

## Proposed change

1. Add validation-side split metrics mirroring the train side: `val/loss_positive`,
   `val/loss_negative`, `val/n_positive`, `val/n_negative`. Simplest approach: reuse
   the per-sample MSE logic from `_compute_negative_weighted_loss()` inside the
   `validation_step`s — but **without** applying `negative_loss_weight` (validation
   loss should stay unweighted for comparability; only the split breakdown is wanted).
2. Add the negative/positive split keys to `csv_log_keys` so they appear in
   `training_log.csv`.
3. Update `docs/guides/negative-frames.md` (currently lists only `train/*` metrics and
   states negatives just "contribute to `val/loss`").

## Repro

```bash
# 1. Build/obtain a project (.slp) that has negative frames:
python -c "
import sleap_io as sio
labels = sio.load_slp('labels.slp')
v = labels.videos[0]
for idx in range(0, 500, 50):
    lf = sio.LabeledFrame(video=v, frame_idx=idx)
    lf.is_negative = True
    labels.labeled_frames.append(lf)
sio.save_slp(labels, 'labels_with_negatives.slp')
"

# 2. Train a bottom-up or single-instance model with negative frames enabled:
#    data_config:
#      use_negative_frames: true
#      negative_loss_weight: 1.0

# 3. Inspect logged metrics (ZMQ feed, WandB/TensorBoard, or trainer.callback_metrics):
#    - train/loss_negative, train/n_negative  -> present
#    - val/loss_negative,  val/n_negative     -> ABSENT   <-- this issue

# 4. Inspect training_log.csv:
#    - no *_negative / *_positive / n_* columns          <-- secondary CSV gap
```

## Code anchors

Current `main` (~commit `0fdb31de` — confirm):

- `sleap_nn/training/lightning_modules.py` — `_compute_negative_weighted_loss()`
  (~411-494, train-only metrics); `validation_step` methods use plain `nn.MSELoss()`
  (~817, ~1333, ~1655, ~2052).
- `sleap_nn/training/model_trainer.py` — `csv_log_keys` (~895-942).
- `docs/guides/negative-frames.md` — metrics section (~118-126, ~167).

Part of shipping negative frames end-to-end (talmolab/sleap#640; backend feature
PR #484).
